src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorage.cs

```diff
@@ -314,18 +315,21 @@ private object ConvertFromStorageFormat(byte[] contents, Type stateType)
 
         public async IAsyncEnumerable<StorageEntry> GetAll([EnumeratorCancellation] CancellationToken cancellationToken)
```

Comment:
It would have been nice to skip alredy migrated entries, but I guess it's ok since we only get metadata? But we still create a new entry everytime

Reply:
> you read the entire entry everytime

that is correct. I did not test with the large collection (it is basically like the implementation existed with AzureBlobStorage - I just added loading the metadata). So you want to do something like that?
```
var metadata = client.LoadMetadata(entry);
if (metadata is null || metadata.IsNotMigrated())
{
    var blob = client.LoadBlob();
    yield return blob;
}
```

Reply:
rename the method to `GetNonMigrated` entires or smth like that

Reply:
rename the method to `GetNonMigrated` entires or smth like that + only fetch the whole entry if it is not migrated (as proposed in previous comment)